app/src/main/java/com/duckduckgo/app/browser/mediaplayback/MediaPlayback.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.mediaplayback
+
+import androidx.core.net.toUri
+import com.duckduckgo.app.browser.mediaplayback.store.MediaPlaybackRepository
+import com.duckduckgo.common.utils.baseHost
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface MediaPlayback {
+    fun isAutoplayEnabledForUrl(url: String): Boolean
+}
+
+@ContributesBinding(AppScope::class)
+class RealMediaPlayback @Inject constructor(
+    private val mediaPlaybackFeature: MediaPlaybackFeature,
+    private val mediaPlaybackRepository: MediaPlaybackRepository,
+) : MediaPlayback {
+
+    override fun isAutoplayEnabledForUrl(url: String): Boolean {
+        val uri = url.toUri()
+        return mediaPlaybackFeature.self()
+            .isEnabled() && mediaPlaybackRepository.exceptions.firstOrNull { it.domain == uri.baseHost } != null
```

Comment:
@anikiki will correct me if I am wrong but I don't think we're using `exceptions` to enable the feature. `exceptions` are used to set `mediaPlaybackRequiresUserGesture` to `false` for those sites where that is required.

@jonathanKingston maybe I misunderstood your comment?

Reply:
The understanding I have of the code is:

`isAutoPlayEnabledForUrl` is `true` when `mediaPlaybackRequiresUserGesture` is enabled and the site exists in the exceptions list in the remote config.
- `webView.settings.mediaPlaybackRequiresUserGesture` is set to `false` when this method is true.

So far, features are just disabled when a site is in the list. For similar cases we have generally used a different key other than exceptions; although I don't think it matters all that much.

The confusion appears to me between the naming of this method and the inverse feature name to enable.
- I think the simplest fix here would be to rename `isAutoplayEnabledForUrl` to something like  `doesMediaPlaybackRequireUserGestureForUrl` and remove the `!` at the callsite.

See the config change: https://github.com/duckduckgo/privacy-configuration/pull/1767/files (I think for this code to work the status would have to be enabled also)

Reply:
OK, so we're talking about naming then. Yeah `doesMediaPlaybackRequireUserGestureForUrl` seems better naming to me too

Reply:
@jonathanKingston I can make the proposed name change along with moving the negation inside the method, but I don't think it actually addresses the issue, which, in my opinion, is controlling something only through exceptions. 

Curious about your statement _"For similar cases we have generally used a different key other than exceptions; although I don't think it matters all that much."_. Do you have an example?

Reply:
Like I said on MM, I think there are two parts here: the remote config and the internal naming.

Remote config: `mediaPlaybackRequiresUserGesture` imo is the correct name should be enabled by default and we only want to disable it for the websites in the exceptions list. Having `disabled` in https://github.com/duckduckgo/privacy-configuration/pull/1767/files is wrong because it will disable the feature for all websites.

For the actual method `isAutoplayEnabledForUrl`, as @aitorvs said changing it to `doesMediaPlaybackRequireUserGestureForUrl` I think makes it less confusing because autoplay is the opposite of require gesture.

Reply:
> Curious about your statement "For similar cases we have generally used a different key other than exceptions; although I don't think it matters all that much.". Do you have an example?

For expressing different states we've used site arrays in settings like customUserAgent: https://github.com/duckduckgo/privacy-configuration/blob/main/overrides/android-override.json#L218-L239

I don't mind too much what we call the feature, just the concern I had was I immediately read the code and config and concluded it was enabling it.